libvast/vast/format/json.hpp

```diff
@@ -223,18 +224,18 @@ caf::error reader<Selector>::read_impl(size_t max_events, size_t max_slice_size,
   size_t produced = 0;
   table_slice_builder_ptr bptr = nullptr;
   bool timeout = false;
-  auto next_line = [&] {
+  auto next_line = [&, start = std::chrono::steady_clock::now()] {
+    auto now = std::chrono::steady_clock::now();
+    if (start + read_timeout_ < now)
```

Comment:
```suggestion
    auto remaining_time = start + read_timeout_ - now;
    if (remaining_time < 0)
```

Reply:
The `rhs` has to be `steady_clock::duration::zero()` for this. Made the change.